packages/quantic/force-app/solutionExamples/main/lwc/exampleInsightPanel/e2e/exampleInsightPanel.e2e.ts

```diff
@@ -0,0 +1,104 @@
+import {testInsight, expect} from './fixture';
+
+let test = testInsight;
+
+test.describe('Example Insight Panel E2E Tests', () => {
+  test.describe('when loading the Insight Panel', () => {
+    test('should render correctly and load results automatically', async ({
+      insightPanel,
+    }) => {
+      expect(insightPanel.insightPanel).toBeVisible();
+      expect(insightPanel.searchbox).toBeVisible();
+      expect(insightPanel.refineToggle).toBeVisible();
+      expect(insightPanel.insightSummary).toBeVisible();
+      expect(insightPanel.foldedResultList).toBeVisible();
+      expect(insightPanel.tabBar).toBeVisible();
+      expect(insightPanel.pager).toBeVisible();
+
+      const firstResult = insightPanel.results.first();
+      expect(firstResult).toBeVisible();
+      expect(await firstResult.textContent()).not.toBeNull();
+
+      const resultsCount = insightPanel.results.count();
+      expect(await resultsCount).toBeGreaterThan(0);
+    });
+  });
+
+  test.describe('the happy path:', () => {
+    test('should allow the user to search, select a facet, change the tab, change the page and clear the filters', async ({
+      insightPanel,
+      search,
+    }) => {
+      // Trigger a search
+      const exampleQuery = 'Test';
+      const searchRequestPromise = search.waitForSearchRequest();
+
+      await insightPanel.searchboxInput.fill(exampleQuery);
+      await insightPanel.searchboxInput.press('Enter');
+
+      await searchRequestPromise;
+      const searchRequestBody = (await searchRequestPromise).postDataJSON();
+      expect(searchRequestBody?.q).toBe(exampleQuery);
+
+      await expect(insightPanel.insightSummary).toContainText(exampleQuery);
+
+      // Change the tab
+      const expectedTabName = 'Articles';
+      const tabClickSearchRequestPromise = search.waitForSearchRequest();
+
+      const secondTab = await insightPanel.getTabByIndex(1);
+      await secondTab.click();
+
+      const searchRequestBodyAfterTabClick = (
+        await tabClickSearchRequestPromise
+      ).postDataJSON();
+      expect(searchRequestBodyAfterTabClick?.tab).toBe(expectedTabName);
```

Comment:
```suggestion
      const expectedTabName = 'Articles';
      const tabClickSearchRequestPromise = search.waitForSearchRequest();
      const secondTab = await insightPanel.getTabByIndex(1);
      await secondTab.click();
      const searchRequestBodyAfterTabClick = (
        await tabClickSearchRequestPromise
      ).postDataJSON();
      expect(searchRequestBodyAfterTabClick?.tab).toBe(expectedTabName);
```

Reply:
Same for everywhere, unless there's a good reason to do so, you can keep the test and expects together.

Reply:
it was for readability but sure